pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -186,6 +197,31 @@ func (c *K8sOrchestrator) InitTopologyServiceInController(ctx context.Context) (
 					csiNodeTopologyInformer:   *crInformer,
 					clusterFlavor:             clusterFlavor,
 					isCSINodeIdFeatureEnabled: c.IsFSSEnabled(ctx, common.UseCSINodeId),
+					isTopologyPreferentialDatastoresFSSEnabled: c.IsFSSEnabled(ctx,
+						common.TopologyPreferentialDatastores),
+				}
+
+				if controllerVolumeTopologyInstance.isTopologyPreferentialDatastoresFSSEnabled {
+					// Get CNS config.
+					cnsCfg, err := common.GetConfig(ctx)
+					if err != nil {
+						return nil, logger.LogNewErrorf(log, "failed to fetch CNS config. Error: %+v", err)
+					}
+					// Fetch preferred datastores and store in cache at regular intervals.
+					go func() {
+						ctx, log := logger.GetNewContextWithLogger()
```

Comment:
Can you move this context creation into for loop?
So we can trace what happened in each refresh cycle?

Reply:
Done